File: app12_kotlin_android/data/src/main/kotlin/com/kotlinguard/data/seeding/SeedDTOs.kt
package com.kotlinguard.data.seeding

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class FrameworkSeed(
    val code: String,
    val name: String,
    val version: String,
    val description: String,
    val referenceUrl: String
)

@Serializable
data class ThreatSeed(
    val frameworkCode: String,
    val code: String,
    val title: String,
    val severity: String,
    val category: String,
    val description: String,
    @SerialName("attack_vector") val attackVector: String,
    @SerialName("attack_surface") val attackSurface: String,
    val stride: String,
    val tags: List<String>
)

@Serializable
data class ThreatTranslationSeed(
    val code: String,
    val locale: String,
    val title: String,
    val description: String,
    @SerialName("attack_vector") val attackVector: String
)

@Serializable
data class CrossReferenceSeed(
    val sourceCode: String,
    val targetCode: String,
    val relationshipType: String,
    val description: String
)

@Serializable
data class MitigationSeed(
    val slug: String,
    val threatCode: String? = null,
    val cardId: String? = null,
    val title: String,
    val description: String,
    val mitigationType: String,
    val effort: String,
    val effectiveness: String
)

@Serializable
data class CodeSampleManifestEntry(
    val mitigationSlug: String,
    val language: String,
    val sampleType: String,
    val title: String,
    val description: String,
    val file: String,
    val frameworkHint: String,
    val versionNote: String
)

File: app11_swift_ios/SwiftGuardData/Sources/SwiftGuardData/Seeding/SeedDTOs.swift
import Foundation

struct FrameworkSeed: Decodable {
    let code: String
    let name: String
    let version: String
    let description: String
    let referenceUrl: String
}

struct ThreatSeed: Decodable {
    let frameworkCode: String
    let code: String
    let title: String
    let severity: String
    let category: String
    let description: String
    let attackVector: String
    let attackSurface: String
    let stride: String
    let tags: [String]

    enum CodingKeys: String, CodingKey {
        case frameworkCode, code, title, severity, category, description, stride, tags
        case attackVector = "attack_vector"
        case attackSurface = "attack_surface"
    }
}

struct ThreatTranslationSeed: Decodable {
    let code: String
    let locale: String
    let title: String
    let description: String
    let attackVector: String

    enum CodingKeys: String, CodingKey {
        case code, locale, title, description
        case attackVector = "attack_vector"
    }
}

struct CrossReferenceSeed: Decodable {
    let sourceCode: String
    let targetCode: String
    let relationshipType: String
    let description: String
}

struct MitigationSeed: Decodable {
    let slug: String
    let threatCode: String?
    let cardId: String?
    let title: String
    let description: String
    let mitigationType: String
    let effort: String
    let effectiveness: String
}

struct CodeSampleManifestEntry: Decodable {
    let mitigationSlug: String
    let language: String
    let sampleType: String
    let title: String
    let description: String
    let file: String
    let frameworkHint: String
    let versionNote: String
}
